frontend/test/pytest/test_from_plxpr.py

```diff
@@ -0,0 +1,531 @@
+        assert qml.math.allclose(catalyst_res[0], expected_expval_x)
+        assert qml.math.allclose(catalyst_res[1], expected_expval_y)
+        assert qml.math.allclose(catalyst_res[2], expected_probs)
+
+        qjit_obj = qml.qjit(circuit)
+        qjit_obj(x, y, z)
+        catalxpr = qjit_obj.jaxpr
+        call_jaxpr_pl = converted.eqns[0].params["call_jaxpr"]
+        call_jaxpr_c = catalxpr.eqns[0].params["call_jaxpr"]
+
+        compare_call_jaxprs(call_jaxpr_pl, call_jaxpr_c)
+
+
+class TestHybridPrograms:
+    """from_plxpr conversion tests for hybrid programs."""
+
+    def test_pre_post_processing(self):
+        """Test converting a workflow with pre and post processing."""
+
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(x, y):
+            qml.RX(x, 0)
+            qml.RY(3 * y + 1, 1)
+            qml.CNOT((0, 1))
+            return qml.expval(qml.X(1)), qml.expval(qml.Y(0))
+
+        def workflow(z):
+            a, b = circuit(z, 2 * z)
+            return a + b
+
+        qml.capture.enable()
+        plxpr = jax.make_jaxpr(workflow)(0.5)
+        qml.capture.disable()
+
+        converted = from_plxpr(plxpr)(0.5)
+
+        res = catalyst_execute_jaxpr(converted)(0.5)
+
+        x = 0.5
+        y = 3 * 2 * 0.5 + 1
+
+        expval_x1 = np.sin(y)
+        expval_y0 = -np.sin(x) * np.sin(y)
+        expected = expval_x1 + expval_y0
+
+        assert qml.math.allclose(expected, res[0])
+
+        qjit_obj = qml.qjit(workflow)
+        qjit_obj(0.5)
+
+        call_jaxpr_pl = converted.eqns[1].params["call_jaxpr"]
+        call_jaxpr_c = qjit_obj.jaxpr.eqns[1].params["call_jaxpr"]
+
+        # qubit extraction and classical equations in a slightly different order
+        # thus cant check specific equations and have to discard comparing counts
+        compare_call_jaxprs(call_jaxpr_pl, call_jaxpr_c, skip_eqns=(4, 5, 6))
+        compare_eqns(call_jaxpr_pl.eqns[4], call_jaxpr_c.eqns[5])
+        compare_eqns(call_jaxpr_pl.eqns[5], call_jaxpr_c.eqns[6])
+        compare_eqns(call_jaxpr_pl.eqns[6], call_jaxpr_c.eqns[4])
+
+    def test_multiple_qnodes(self):
+        """Test that a workflow with multiple qnodes can be converted."""
+
+        @qml.qnode(qml.device("lightning.qubit", wires=1))
+        def f(x):
+            qml.RX(x, wires=0)
+            return qml.expval(qml.Y(0))
+
+        @qml.qnode(qml.device("lightning.qubit", wires=2))
+        def g(y):
+            qml.Hadamard(0)
+            qml.IsingXX(y, wires=(0, 1))
+            return qml.expval(qml.PauliZ(1))
+
+        def workflow(x, y):
+            return f(x) + g(y)
+
+        qml.capture.enable()
+        jaxpr = jax.make_jaxpr(workflow)(0.5, 1.2)
+        qml.capture.disable()
+        catalxpr = from_plxpr(jaxpr)(0.5, 1.2)
+        results = catalyst_execute_jaxpr(catalxpr)(0.5, 1.2)
+
+        expected = -np.sin(0.5) + np.cos(1.2)
+
+        assert qml.math.allclose(results, expected)
```

Comment:
```suggestion
        assert qml.math.allclose(results, expected)


if __name__ == "__main__":
    pytest.main(["-x", __file__])

```

I like to have this little block in our test files because it allows you to just hit the debug button in vscode to run your tests through the debugger, or run the tests by executing the module as a script :)